include/ginkgo/core/base/index_set.hpp

```diff
@@ -196,6 +218,28 @@ class IndexSet : public EnablePolymorphicObject<IndexSet<IndexType>> {
      */
     index_type get_local_index(index_type global_index) const;
 
+    /**
+     * Return which set the global index belongs to.
+     *
+     * Consider the set idx_set = (0, 1, 2, 4, 6, 7, 8, 9). This function
+     * returns the subset id in the index set of the input global index. For
+     * example, `idx_set.get_subset_id(0) == 0` `idx_set.get_subset_id(4)
+     * == 1` and `idx_set.get_subset_id(6) == 2`.
+     *
+     * @note This function returns a scalar value and needs a scalar value.
+     *       For repeated queries, it is more efficient to use the Array
+     *       functions that take and return arrays which allow for more
+     *       throughput.
+     *
+     * @param global_index  the global index.
+     *
+     * @return  the local index of the element in the index set.
+     *
+     * @warning This single entry query can have significant kernel lauch
+     *          overheads and should be avoided if possible.
+     */
+    index_type get_subset_id(index_type global_index) const;
```

Comment:
Do we need this functionality? Its performance is even worse than get_local_index and it's not used anywhere outside tests

Reply:
Yes, it is useful to implement some quick checks and prototyping and I agree that this is inefficient and should not be used in performance critical areas, and hence the warning.